index.d.ts

```diff
@@ -195,14 +196,26 @@ declare namespace meow {
 		readonly hardRejection?: boolean;
 	}
 
-	type TypedFlags<Flags extends AnyFlags> = {
-		[F in keyof Flags]: Flags[F] extends {type: 'number'}
+	type TypedFlag<Flag extends AnyFlag> =
+		Flag extends {type: 'number'}
 			? number
-			: Flags[F] extends {type: 'string'}
+			: Flag extends {type: 'string'}
 				? string
-				: Flags[F] extends {type: 'boolean'}
+				: Flag extends {type: 'boolean'}
 					? boolean
 					: unknown;
+
+	type PossiblyOptionalFlag<Flag extends AnyFlag> =
+		Flag extends {isRequired: true}
+			? TypedFlag<Flag>
+			: Flag extends {default: any}
+				? TypedFlag<Flag>
+				: TypedFlag<Flag>|undefined;
+
+	type TypedFlags<Flags extends AnyFlags> = {
+		[F in keyof Flags]: Flags[F] extends {isMultiple: true}
+			? Array<TypedFlag<Flags[F]>>
```

Comment:
I think this branch should have the same logic as in `PossiblyOptionalFlag`. A flag with `isMultiple: true` but no `default` will return `undefined` for an empty argv, no? I couldn't find a test covering this case, but tried the following locally and it fails:

```
t.deepEqual(meow({
	argv: [],
	flags: {
		fooBar: {
			type: 'string',
			isMultiple: true
		}
	}
}).flags, {
	fooBar: []
});
```

```
  Difference:

    {
  +   fooBar: [],
    }
```

Maybe this could be addressed in another PR, alongside with a test, since this already seems like an improvement.

Reply:
@voxpelli

Reply:
Thanks for the ping @sindresorhus, pushed a fix